server/src/main/java/game/snake/Snake.java

```diff
@@ -109,6 +107,11 @@ public void tick() {
 
         // update chunks
         currentChunk.append(encDirDelta, fast);
+
+        //ensures that the current snakechunk is added to all worldchunks in which the snake exists
+        var currentSnakeHeadIntersectingWorldChunks = world.chunks.findIntersectingChunks(headPosition, getWidth() / 2);
```

Comment:
This variable is only used in one place so I don't see the point. You could make this a one liner.